tajo-storage/tajo-storage-hdfs/src/main/java/org/apache/tajo/storage/FileTablespace.java

```diff
@@ -892,9 +892,8 @@ protected Path commitOutputData(OverridableConf queryContext, boolean changeFile
           boolean movedToOldTable = false;
           boolean committed = false;
           Path oldTableDir = new Path(stagingDir, TajoConstants.INSERT_OVERWIRTE_OLD_TABLE_NAME);
-          ContentSummary summary = fs.getContentSummary(stagingResultDir);
 
-          if (!queryContext.get(QueryVars.OUTPUT_PARTITIONS, "").isEmpty() && summary.getFileCount() > 0L) {
+          if (!queryContext.get(QueryVars.OUTPUT_PARTITIONS, "").isEmpty()) {
```

Comment:
It would be better if it works according to session variable to allow users to decide whether original data set is removed or not when inserting rows is empty.